src/Data/List/Properties.agda

```diff
@@ -624,6 +626,47 @@ length-replicate : ∀ n {x : A} → length (replicate n x) ≡ n
 length-replicate zero    = refl
 length-replicate (suc n) = cong suc (length-replicate n)
 
+lookup-replicate : ∀ n (x : A) (i : Fin (length (replicate n x))) → lookup (replicate n x) i ≡ x
+lookup-replicate (suc n) x zero    = refl
+lookup-replicate (suc n) x (suc i) = lookup-replicate n x i
+
+map-replicate :  ∀ (f : A → B) (x : A) n →
+                 map f (replicate n x) ≡ replicate n (f x)
+map-replicate f x zero    = refl
+map-replicate f x (suc n) = cong (_ ∷_) (map-replicate f x n)
+
+zipWith-replicate : ∀ n (_⊕_ : A → B → C) (x : A) (y : B) →
+                    zipWith _⊕_ (replicate n x) (replicate n y) ≡ replicate n (x ⊕ y)
+zipWith-replicate zero    _⊕_ x y = refl
+zipWith-replicate (suc n) _⊕_ x y = cong (x ⊕ y ∷_) (zipWith-replicate n _⊕_ x y)
+
+------------------------------------------------------------------------
+-- iterate
+
+length-iterate : ∀ n {f} {x : A} → length (iterate f x n) ≡ n
+length-iterate zero    = refl
+length-iterate (suc n) = cong suc (length-iterate n)
+
+toList-iterate : ∀ n {f} {x : A} → toList (Vec.iterate f x {n = n}) ≡ iterate f x n
+toList-iterate zero    = refl
+toList-iterate (suc n) = cong (_ ∷_) (toList-iterate n)
+
+iterate-id : ∀ n {x : A} → iterate id x n ≡ replicate n x
+iterate-id zero    = refl
+iterate-id (suc n) = cong (_ ∷_) (iterate-id n)
+
+take-iterate : ∀ n {f} {x : A} → take n (iterate f x n) ≡ iterate f x n
```

Comment:
Is this not a (fairly immediate?) consequence of `length-iterate` and `take-all`?

Even if it is worth retaining as a lemma in its own right, please consider bnding `let xs = iterate f x n in ...` in the type, as in L629 above

Reply:
The lemma is probably worth retaining, but I agree that it would be nice to have the proof use those results instead of being done "by hand", even though the resulting proof is quite simple.

Reply:
Should it be written like this instead? -
```agda
drop-iterate : ∀ n f (x : A) → drop n (iterate f x n) ≡ []
drop-iterate n f x = drop-all n (iterate f x n) {!(length-iterate n f x)!}
```

Reply:
```agda
take-iterate : ∀ n {f} {x : A} → let xs = iterate f x n in take n xs ≡ xs
take-iterate n {f} {x} = take-all n xs n≥length[xs]
  where
  xs = iterate f x n
  n≥length[xs] : n ≥ length xs
  n≥length[xs] rewrite length-iterate n {f} {x} = ≤-refl
```
NB You are going to have a hard time inferring `f` in these proofs (Agda can't unify functions easily), so, as here, the arguments will (probably: CHECK!) need to be passed explicitly, so nothing to be gained by keeping them implicit here?

Reply:
See my review comment, requesting changes, below.